crates/cdk/src/wallet/melt.rs

```
@@ -313,14 +314,32 @@ impl Wallet {
             .map(|k| k.id)
             .collect();
         let keyset_fees = self.get_keyset_fees().await?;
-        let input_proofs = Wallet::select_proofs(
+        let (mut input_proofs, mut exchange) = Wallet::select_exact_proofs(
             inputs_needed_amount,
             available_proofs,
             &active_keyset_ids,
             &keyset_fees,
             true,
         )?;
 
+        if let Some((proof, exact_amount)) = exchange.take() {
+            if let Ok(Some(new_proofs)) = self
```

I think we should error here and not silently fail.